quic-client/src/nonblocking/quic_client.rs

```diff
@@ -117,7 +118,14 @@ impl QuicLazyInitializedEndpoint {
         };
 
         let mut crypto = rustls::ClientConfig::builder()
-            .with_safe_defaults()
+            .with_cipher_suites(&[
+                TLS13_AES_128_GCM_SHA256,
+                TLS13_CHACHA20_POLY1305_SHA256,
+                TLS13_AES_256_GCM_SHA384,
+            ])
+            .with_kx_groups(&ALL_KX_GROUPS)
+            .with_protocol_versions(&[&TLS13])
+            .expect("Failed to create TLS 1.3 client")
```

Comment:
let's dereplicode with something like

```rust
trait SolanaQuicCryptoConfig {
    fn with_solana_quic_crypto_config(self) -> Self;
}

impl SolanaQuicCryptoConfig for rustls::ConfigBuilder {
    fn with_solana_quic_crypto_config(self) -> Self {
        self
            .with_cipher_suites(&[
                TLS13_AES_128_GCM_SHA256,
                TLS13_CHACHA20_POLY1305_SHA256,
                TLS13_AES_256_GCM_SHA384,
            ])
            .with_kx_groups(&ALL_KX_GROUPS)
            .with_protocol_versions(&[&TLS13])
            .expect("Failed to create TLS 1.3 client")
    }
}
```

then the usage changes just look like

```diff
-            .with_safe_defaults()
+            .with_solana_quic_crypto_config()
```